padertorch/configurable.py

```diff
@@ -1348,9 +1456,9 @@ def keys(self):
         return tuple(self.data.keys())
 
     def _check_redundant_keys(self, msg):
-        assert 'factory' in self.data
-        factory = import_class(self.data['factory'])
-        parameters = inspect.signature(factory).parameters.values()
+        assert self.special_key in self.data, self.data
```

Comment:
`None in {None: ...}` returns `True`.
While this is unlikely, `assert self.special_key is not None, f'Missing factory or partial in {self.data}'` does the correct check.

Reply:
Or `assert self.special_key, f'Missing factory or partial in {self.data}'` as you used at other positions.